src/Data/Items/LootTableEntry.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+
+namespace ThirdRun.Items
+{
+    /// <summary>
+    /// Represents an entry in a monster's loot table with weight and generation logic
+    /// </summary>
+    public abstract class LootTableEntry
+    {
+        public int Weight { get; }
+
+        protected LootTableEntry(int weight)
+        {
+            Weight = weight;
+        }
+
+        /// <summary>
+        /// Generate an item based on this loot entry
+        /// </summary>
+        public abstract Item GenerateItem(int monsterLevel, Random random);
+    }
+
+    /// <summary>
+    /// Loot entry that generates random items with specified rarity
+    /// </summary>
+    public class RandomLootEntry : LootTableEntry
+    {
+        public ItemTemplate? Template { get; }
+        public ItemRarity Rarity { get; }
+
+        /// <summary>
+        /// Creates a RandomLootEntry that uses a specific template
+        /// </summary>
+        public RandomLootEntry(int weight, ItemTemplate template, ItemRarity rarity) : base(weight)
+        {
+            Template = template ?? throw new ArgumentNullException(nameof(template));
+            Rarity = rarity;
+        }
+
+        /// <summary>
+        /// Creates a RandomLootEntry that randomly selects from all available templates
+        /// </summary>
+        public RandomLootEntry(int weight, ItemRarity rarity) : base(weight)
```

Comment:
Remove this constructor, must not be used. Use the one that takes a ItemTemplate parameter